pkg/manager/member/tiflash_member_manager.go

```diff
@@ -397,13 +397,40 @@ func getNewStatefulSet(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap) (*apps.St
 			},
 		},
 	}
+	script := "set -ex;ordinal=`echo ${POD_NAME} | awk -F- '{print $NF}'`;sed s/POD_NUM/${ordinal}/g /etc/tiflash/config_templ.toml > /data0/config.toml;sed s/POD_NUM/${ordinal}/g /etc/tiflash/proxy_templ.toml > /data0/proxy.toml"
+
+	if len(tc.Spec.ClusterDomain) > 0 {
```

Comment:
Also, add a `TODO` comment here for the case that a heterogeneous cluster without local PD deployed across Kubernetes clusters?

Reply:
https://github.com/pingcap/tidb-operator/pull/3947/commits/5fd52d1762d237990b52d3593727b8789d059edd